internal/security/credentials.go

```diff
@@ -153,5 +164,20 @@ func (s *SecretProvider) updateSecretsCache(path string, secrets map[string]stri
 // path specifies the type or location of the secrets to store
 // secrets map specifies the "key": "value" pairs of secrets to store
 func (s *SecretProvider) StoreSecrets(path string, secrets map[string]string) error {
-	return nil // TODO: returning nil until interface is implemented
+
+	if !s.isSecurityEnabled() {
+		return errors.New("Storing secrets is not supported when running insecure mode")
+	}
+
+	err := s.SecretClient.StoreSecrets(path, secrets)
```

Comment:
Should we check the `secrets` is empty or not?

Reply:
Good catch! I will add at

Reply:
Decided to add that check to the web server right before the call of StoreSecrets

Reply:
right, got you